python/lsst/obs/fiberspectrograph/spectrum.py

```diff
@@ -0,0 +1,198 @@
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = ("FiberSpectrum",)
+
+import numpy as np
+import astropy.io.fits
+import fitsio
+import astropy.units as u
+from ._instrument import FiberSpectrograph
+import lsst.afw.image as afwImage
+
+
+class VisitInfo:
+    def __init__(self, md):
+        self.exposureTime = md["EXPTIME"]
+
+    def getExposureTime(self):
+        return self.exposureTime
+
+
+class Info:
+    def __init__(self, md):
+        self.visitInfo = VisitInfo(md)
+
+    def getVisitInfo(self):
+        return self.visitInfo
+
+
+class FiberSpectrum:
+    def __init__(self, wavelength, flux, md=None):
+        self.wavelength = wavelength
+        self.flux = flux
+        self.md = md
+
+        self.info = Info(md)
+        self.detector = FiberSpectrograph().getCamera()[0]
+
+        self.__Mask = afwImage.MaskX(1, 1)
+        self.getPlaneBitMask = self.__Mask.getPlaneBitMask  # ughh, awful Mask API
+        self.mask = np.zeros(flux.shape, dtype=self.__Mask.array.dtype)
+        self.variance = np.zeros_like(flux)
+
+    def getDetector(self):
+        return self.detector
+
+    def getInfo(self):
+        return self.info
+
+    def getMetadata(self):
+        return self.md
+
+    def getFilter(self):
+        return FiberSpectrograph.filterDefinitions[0].makeFilterLabel()
+
+    def getBBox(self):
+        return self.detector.getBBox()
+
+    @staticmethod
+    def readFits(path):
+        """Read a Spectrum from disk"
+
+        Parameters
+        ----------
+        path : `str`
+            The file to read
+
+        Returns
+        -------
+        spectrum : `~lsst.obs.fiberspectrograph.FiberSpectrum`
+            In-memory spectrum.
+        """
+        md = dict(fitsio.read_header(path))
+        flux = fitsio.read(path)
+        wavelength = fitsio.read(path, ext=md["PS1_0"], columns=md["PS1_1"]).flatten()
+
+        wavelength = u.Quantity(wavelength, u.Unit(md["CUNIT1"]), copy=False)
+
+        return FiberSpectrum(wavelength, flux, md)
+
+    def writeFits(self, path):
+        """Write a Spectrum to disk
+
+        Parameters
+        ----------
+        path : `str`
+            The file to write
+        """
+        hdl = DataManager(self).make_hdulist()
+
+        hdl.writeto(path)
```

Comment:
Can you confirm that the `DATE` header does reflect the date the HDU was written out?

Reply:
Here an example of an output file's header. 

<img width="816" alt="Screenshot 2024-02-09 at 12 46 52 PM" src="https://github.com/lsst/obs_fiberspectrograph/assets/12816116/d43e286c-1eb4-4806-8d53-af12445ca009">

Reply:
Those are added by the metadata translator. Not by the FITS writer. FITS writers are meant to add a `DATE` header to the HDU indicating when the HDU was created. Maybe the data_manager class needs to be fixed here and in ts_fiberspectrograph.